cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LineageHTTPHandler.java

```diff
@@ -115,6 +116,27 @@ public void datasetLineage(HttpRequest request, HttpResponder responder,
                          lineage, getCollapseTypes(collapse)), LineageRecord.class));
   }
 
+  /**
+   * Get the lineage information about a dataset. This method does not give any information on field level lineage.
+   *
+   * @param namespaceId the namespace in which the pipeline was created
+   * @param appId the appID
+   * @param workflowName the name of the workflow
+   * @param runId the program run ID
+   */
+  @GET
+  @Path("/namespaces/{namespace-id}/apps/{app-id}/workflows/{workflow-name}/runs/{run-id}/endpoints")
```

Comment:
this shouldn't just be for workflows. This endpoint also should be moved to ProgramLifecycleHttpHandler with the other run APIs

Reply:
Offline discussion summary: modified the path. Will keep it in the lineage handler even though the endpoints are for a program run as other APIs reading from lineage tables is in this file